core/src/main/java/hudson/util/io/TarArchiver.java

```diff
@@ -74,6 +74,7 @@ public void visitSymlink(File link, String target, String relativePath) throws I
         tar.putArchiveEntry(e);
         tar.closeArchiveEntry();
         entriesWritten++;
+        files.add(link.getAbsolutePath());
```

Comment:
```suggestion
        files.add(relativePath);
```

Reply:
the problem with this relative path is that depends on where the stash is done.
If you stash from the root level in the pipeline, is ok.  but if you stash from a directory, you will only get the filename.
Therefore, the only way to display the relative path is be calculated in the StashStep by using the job name and its parent.

Reply:
> if you stash from a directory, you will only get the filename

Well, a relative path from that directory. So

```groovy
sh 'mvn verify'
dir('target') {
  stash name: 'stuff', includes: '**/*.txt'
}
```

will print, say,

```
Stashing surefire-reports/TEST-something.txt
```

when it is processing `/home/jenkins/workspace/myjob/target/surefire-reports/TEST-something.txt`. Which is exactly what it should do.

Reply:
I'd rather go for `stashing myjob/target/surefire-reports/TEST-something.txt`

Reply:
You cannot. There is no distinguished “root directory”. Steps should pay attention to the `FilePath` given to them in `StepContext` and nothing else.

Reply:
I agree with Jesse here, I think it makes the most sense to print the relative paths.